### ISSUE TITLE
Error during building api.

### ISSUE THREAD

**Issue by phreakus** [codexgigassys/codex-backend#1]
When I execute "sudo docker-compose up" command, httpd and db execute normally, but during building api some error is occured.
- OS : Ubuntu 16.04 LTS
- Python : 2.7.12
- Error message is like below.
  **Errors were encountered while processing:**
  python-support
  python-cairo
  python-gtk2
  python-glade2
  python-ndg-httpsclient
  **E: Sub-process /usr/bin/dpkg returned an error code (1)**
  **ERROR: Service 'api' failed to build:** The command '/bin/sh -c apt-get clean && apt-get update -qq && apt-get install -y build-essential libpq-dev python-hachoir-metadata p7zip-full libffi-dev libssl-dev python-dev libfuzzy-dev python-numpy python-scipy python-matplotlib python-gevent python-pip python-magic python-crypto zip python-dateutil python-mysqldb redis-server autoconf openssl file python git autoconf automake libc-dev libtool python-dev unzip libfreetype6-dev libtaoframework-freetype-cil-dev libxft-dev' returned a non-zero code: 100

Do you have any idea?

**Comment by CrimsonGlory** [comment]
Hi, can you provide us the output of the following commands?
`# docker version`
`# docker-compose --version`
`# docker images`

**Comment by sashlabs-cyber** [comment]
Hi CrimsonGlory,

I came across the same error, and please find the details which you requested.

Regards,

Sai Santosh
![image](https://cloud.githubusercontent.com/assets/13641169/17552706/08c6f534-5f1f-11e6-8c72-1bb3a136925b.png)

**Comment by CrimsonGlory** [comment]
I have the same versions of everything and I can't reproduce the issue.  I think this may be related to [docker](https://docs.docker.com/engine/userguide/networking/default_network/configure-dns/) [dns](https://docs.docker.com/engine/userguide/networking/configure-dns/) config.

Can you try to edit `docker-compose.yml`file and add the following line at the end of the file:

```
  dns: 8.8.4.4
```

The result will look like this:

```
db:
  image: mongo:3.2
  volumes:
    - ../mongo-data/:/data/db

httpd:
    image: httpd:2.4
    volumes:
        - ../codex-frontend/:/usr/local/apache2/htdocs/
    ports:
        - "6100:80"
api:
  build: .
  command: bash -c "(echo updating pip packages && pip install -r /myapp/src/pip_requeriments.txt && echo starting redis && service redis-server start && (echo starting redis worker default && rq worker --path /myapp/src/ default &) && (echo starting redis worker process && rq worker --path /myapp/src/ process &) && (echo starting redis worker checkup && rq worker --path /myapp/src/MalwareCheckUp/ checkup &)) && echo starting api...Check 127.0.0.1:8080 && python /myapp/src/api2.py -H $$HOSTNAME"
  volumes:
    - .:/myapp
  ports:
    - "4500:8080"
  links:
    - db
    - httpd
  dns: 8.8.4.4
```

Then retry:

```
docker-compose up --build
```

**Comment by sashlabs-cyber** [comment]
Hi CrimsonGlory,

I have tried the way you suggested. Unfortunately it didn't work even now.

![image](https://cloud.githubusercontent.com/assets/13641169/17641918/a5dde6e2-6150-11e6-86a9-08a3751d53d1.png)

**Comment by wigpenepel** [comment]
I have a similar issue, I build it on VM Workstation with Ubuntu Server 16.04.1 first with a 20Gb disk, this was a mistake, however it did work fully, until I dumped several Gig of malware on there and ran out of disk space.
Tore it down and rebuilt with a 250Gb allocation, I am suffering the same issue. Without rectification.  The build times were only 2 hours apart with no other changes.
Whilst I know this answer is not helpful at this time, it will be when I work out how to fix it, luckily I have my -vv switch enabled today and will fully document any solution I find.

**Comment by ror13h** [comment]
**tl;dr** Replace '_pylab_' in _codex-backend/src/pip_requeriments.txt_ with '_matplotlib_'

VM: Ubuntu 16.04 LTS; Docker 1.12.1; Docker-Compose 1.8.0

Same API build error:
![capture](https://cloud.githubusercontent.com/assets/17208129/18518288/6a3129a2-7a6d-11e6-8358-536beb9d7e85.PNG)
Cause seems to be a missing/renamed python lib in pip_requeriments.txt
![capture2](https://cloud.githubusercontent.com/assets/17208129/18518299/6f5c2f8a-7a6d-11e6-9f21-043b5d770017.PNG)
Replacing '_pylab_' with '_matplotlib_' seems to satisfy the missing lib. 
Re-run _docker-compose up_ and all should work. 

Thanks for releasing this tool-- looking forward to exploring it.

**Comment by CrimsonGlory** [comment]
Thanks everyone for the feedback.
It seems we have multiple issues here.

The s/pylab/matplotlib commit has been pushed. But what is happening to wigpenepel, cyber-cyborg999, phreakus seems to be a different thing, because it fails in the first step of the Docker build.
I added more verbose output to apt-get to see if we can find out whats going on.
For the time being I would recommend to use the ready-to-use Codex VM's.
